x-pack/packages/kbn-elastic-assistant/impl/assistant/chat_send/use_chat_send.tsx

```diff
@@ -21,51 +23,47 @@ import { getDefaultSystemPrompt, getDefaultNewSystemPrompt } from '../use_conver
 export interface UseChatSendProps {
   allSystemPrompts: PromptResponse[];
   currentConversation?: Conversation;
-  editingSystemPromptId: string | undefined;
+  currentSystemPromptId: string | undefined;
   http: HttpSetup;
+  refetchCurrentUserConversations: DataStreamApis['refetchCurrentUserConversations'];
   selectedPromptContexts: Record<string, SelectedPromptContext>;
-  setEditingSystemPromptId: React.Dispatch<React.SetStateAction<string | undefined>>;
+  setCurrentSystemPromptId: React.Dispatch<React.SetStateAction<string | undefined>>;
   setSelectedPromptContexts: React.Dispatch<
     React.SetStateAction<Record<string, SelectedPromptContext>>
   >;
-  setUserPrompt: React.Dispatch<React.SetStateAction<string | null>>;
   setCurrentConversation: React.Dispatch<React.SetStateAction<Conversation | undefined>>;
 }
 
 export interface UseChatSend {
   abortStream: () => void;
   handleOnChatCleared: () => Promise<void>;
-  handlePromptChange: (prompt: string) => void;
-  handleSendMessage: (promptText: string) => void;
   handleRegenerateResponse: () => void;
+  handleChatSend: (promptText: string) => Promise<void>;
+  setUserPrompt: React.Dispatch<React.SetStateAction<string | null>>;
   isLoading: boolean;
+  userPrompt: string | null;
 }
 
 /**
- *  handles sending messages to an API and updating the conversation state.
- *  Provides a set of functions that can be used to handle user input, send messages to an API,
- *  and update the conversation state based on the API response.
+ * Handles sending user messages to the API and updating the conversation state.
  */
 export const useChatSend = ({
   allSystemPrompts,
   currentConversation,
-  editingSystemPromptId,
+  currentSystemPromptId,
   http,
+  refetchCurrentUserConversations,
   selectedPromptContexts,
-  setEditingSystemPromptId,
+  setCurrentSystemPromptId,
   setSelectedPromptContexts,
-  setUserPrompt,
   setCurrentConversation,
 }: UseChatSendProps): UseChatSend => {
   const { assistantTelemetry, toasts } = useAssistantContext();
+  const [userPrompt, setUserPrompt] = useState<string | null>(null);
 
   const { isLoading, sendMessage, abortStream } = useSendMessage();
   const { clearConversation, removeLastMessage } = useConversation();
 
-  const handlePromptChange = (prompt: string) => {
```

Comment:
`handlePromptChange` was just a wrapper on top of `setUserPrompt`, so it was eliminated.